### ISSUE TITLE
Usage question: what is the diference between applying a transformation and running a `fit`?

### ISSUE THREAD

**Issue by batata004** [RubixML/ML#96]
What is the practical difference of both codes below? Does the second one change anything in the dataset? On my understanding, these codes should just show the importances of each columns, and nothing else, right?

```
$transformer = new RecursiveFeatureEliminator();
$transformer->fit($dataset);
$importances = $transformer->importances();
```

```
$transformer = new RecursiveFeatureEliminator();
$dataset->apply($transformer);
$importances = $transformer->importances();
```

**Comment by andrewdalpino** [comment]
The second block of code fits the dataset and applies the transformation, the first code block does not (it only fits). Applying the transformation is not necessary if you are only interested in returning feature importances.

The rest of your question can be gleaned from the documentation ...

https://docs.rubixml.com/en/latest/preprocessing.html
https://docs.rubixml.com/en/latest/transformers/api.html

In short, fitting a dataset to a transformer is alot like training a learner ... the transformer builds some internal representation that allows it to carry out future transformations and does not transform any samples in the process. Applying a transformation to a dataset is the operation of both fitting and transforming the samples.

**Comment by batata004** [comment]
Thank you again so much! Very clear information you provided :)
